src/config.rs

```diff
@@ -37,12 +39,14 @@ pub struct ClientConfig {
 /// A builder for a [`Client`].
 pub struct ClientBuilder {
     vendor_endpoint: Url,
+    retry_policy: Option<ExponentialBackoff>,
 }
 
 impl Default for ClientBuilder {
     fn default() -> ClientBuilder {
         ClientBuilder {
             vendor_endpoint: DEFAULT_VENDOR_ENDPOINT.clone(),
+            retry_policy: Some(ExponentialBackoff::builder().build_with_max_retries(5)),
```

Comment:
Can we add another field to the config to allow the max retries to be set by the client?

Reply:
yes! well kind of. 
You can now do the following, which I think isn't too bad to write, and leaves the most freedom for configuring the client.

```rust
let client = Client::builder()
  .with_retry_policy(ExponentialBackoff::builder().build_with_max_retries(10))
  .build(ClientConfig{
    client_id: id,
    client_secret_key: key,
  })
```